copier/jinja_ext.py

```diff
@@ -0,0 +1,106 @@
+"""Jinja2 extensions built for Copier."""
+
+from __future__ import annotations
+
+from typing import Any, Callable, Iterable
+
+from jinja2 import nodes
+from jinja2.exceptions import UndefinedError
+from jinja2.ext import Extension
+from jinja2.parser import Parser
+from jinja2.sandbox import SandboxedEnvironment
+
+
+class YieldEnvironment(SandboxedEnvironment):
```

Comment:
I wonder if it shouldn't be called `CopierEnvironment`, since it's a custom environment for Copier. Would prevent renaming it later if it is customized further. No strong opinion.